### PR TITLE
Change default hadoop version from 2.2.0 to 2.7.1

### PR DESCRIPTION
**Pull request by maobaolong**

### CODE REVIEW

**Comment by maobaolong** [comment]
@apc999 PTAL.

**Comment by gpang** [comment]
@maobaolong @apc999 Would it make sense to target this to master, then cherry-pick to 1.8?

**Comment by apc999** [comment]
@gpang agree. @maobaolong  can you create the same change into master branch? For branch 1.8, I am hesitated to cherry-pick and bump the hadoop version as we typically keep the dependency stable across patch releases (e.g., from 1.8.1 to 1.8.2)

Independently, I agree to bump the default HDFS version to 2.7 from 2.2 in Alluxio 2.0. Latest Spark depends on HDFS 2.7, which will make the src code dependency easy to handle. @calvinjia @gpang , are you good?

Separately, @gpang @calvinjia , I think we should bump

**Comment by apc999** [comment]
@calvinjia , do we want to include this change to 2.0.0? On the binary side, it should make no difference. Only affecting src code level dependency. Right?

File: dev/scripts/src/alluxio.org/build-distribution/cmd/common.go [hunk 1]
```diff
@@ -32,11 +32,11 @@ var hadoopDistributions = map[string]version{
 	"hadoop-2.4": parseVersion("2.4.1"),
 	"hadoop-2.5": parseVersion("2.5.2"),
 	"hadoop-2.6": parseVersion("2.6.5"),
-	"hadoop-2.7": parseVersion("2.7.3"),
+	"hadoop-2.7": parseVersion("2.7.1"),
```
Comment (calvinjia) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/common.go]: Is there a reason we want to use `.1` instead of `.3` or latest (`.7`)?

**Comment by apc999** [comment]
Link issue https://github.com/Alluxio/alluxio/issues/9753 to this PR. We are aiming to solve it in Alluxio v2.1

**Comment by ns1123** [comment]
@maobaolong you will also need to update the default value of the UNDERFS_VERSION constant in PropertyKey.java. It is currently set to "2.2".

**Comment by apc999** [comment]
This PR https://github.com/Alluxio/alluxio/pull/9897/files takes care of the default HDFS version bumping

**Pull request closed without merging**